rotkehlchen/chain/ethereum/modules/the_graph/decoder.py

```diff
@@ -0,0 +1,179 @@
+        return [THE_GRAPH_CPT_DETAILS]
+
+    def possible_events(self) -> DecoderEventMappingType:
+        return {CPT_THE_GRAPH: {
+            HistoryEventType.STAKING: {
+                HistoryEventSubType.DEPOSIT_ASSET: EventCategory.DEPOSIT,
+                HistoryEventSubType.REMOVE_ASSET: EventCategory.WITHDRAW,
+            },
+            HistoryEventType.INFORMATIONAL: {
+                HistoryEventSubType.NONE: EventCategory.INFORMATIONAL,
+            },
+        }}
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            CONTRACT_THE_GRAPH_STAKING: (self._decode_delegator_staking,),
+        }
+
+    # processes events from The Graph's staking contract
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+        token = A_GRT.resolve_to_evm_token()
+        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(token, context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(token, context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(token, context)
+        return DEFAULT_DECODING_OUTPUT
+
+    # processes StakeDelegated event
+    def _decode_stake_delegated(
+            self,
+            token: 'EvmToken',
+            context: DecoderContext,
+    ) -> DecodingOutput:
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        stake_amount = token_normalized_value(
+            token_amount=hex_or_bytes_to_int(context.tx_log.data[:32]),
+            token=token,
+        )
+        # find Transfer event with delegation tax (0.5% GRT burn) and persist the burnt amount
+        tokens_burnt = 0
+        for ctx_log in context.all_logs:
+            if (
+                ctx_log.topics[0] == TOPIC_TRANSFER and
+                hex_or_bytes_to_address(ctx_log.topics[1]) == CONTRACT_THE_GRAPH_STAKING and
+                hex_or_bytes_to_address(ctx_log.topics[2]) == ZERO_ADDRESS
+            ):
+                tokens_burnt = hex_or_bytes_to_int(ctx_log.data)
+        # identify and override the original stake Transfer event
+        for event in context.decoded_events:
+            if (
+                event.location_label == delegator and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.STAKING
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Delegated {stake_amount} GRT to Indexer {indexer}'
```

Comment:
hmmm no no. You actually need a separate event for the burning. You need to use the burning event. And create a spend/fee or something like that event. Otherwise accounting will be off. You send 1000 GRT, and only 995 arrive. The rest are lost and not able to account for. 

Remember `notes` are human readable, but completely ignored when processing events. Events are also supposed to be for 1 asset movement either in or out. And the way you currently do it, we are missing 0.05% of the tokens each time

Reply:
Let's validate I've understood this correctly. In the end we should have:
1) STAKING/DEPOSIT_ASSET event with balance=995 GRT
2) SPEND/FEE event with balance=5 GRT
Is that correct?

Reply:
Updated. There's informational event between STAKING/DEPOSIT_ASSET and [SPEND/FEE](https://github.com/rotki/rotki/pull/6712/files#diff-a44165e5a20fc6ff8b5f163bfcab946d05dd19e62835576c365aa941790408ccR75) now - I'm assuming that also should work?